libs/partners/nvidia-trt/langchain_nvidia_trt/llms.py

```diff
@@ -0,0 +1,405 @@
+        return values
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of LLM."""
+        return "nvidia-trt-llm"
+
+    @property
+    def _model_default_parameters(self) -> Dict[str, Any]:
+        return {
+            "tokens": self.tokens,
+            "top_k": self.top_k,
+            "top_p": self.top_p,
+            "temperature": self.temperature,
+            "repetition_penalty": self.repetition_penalty,
+            "length_penalty": self.length_penalty,
+            "beam_width": self.beam_width,
+        }
+
+    @property
+    def _identifying_params(self) -> Dict[str, Any]:
+        """Get all the identifying parameters."""
+        return {
+            "server_url": self.server_url,
+            "model_name": self.model_name,
+            **self._model_default_parameters,
+        }
+
+    def _get_invocation_params(self, **kwargs: Any) -> Dict[str, Any]:
+        return {**self._model_default_parameters, **kwargs}
+
+    def get_model_list(self) -> List[str]:
+        """Get a list of models loaded in the triton server."""
+        res = self.client.get_model_repository_index(as_json=True)
+        return [model["name"] for model in res["models"]]
+
+    def _get_model_concurrency(self, model_name: str, timeout: int = 1000) -> int:
+        """Get the model concurrency."""
+        # (WFH): This isn't used anywhere...
+        self._load_model(model_name, timeout)
+        instances = self.client.get_model_config(model_name, as_json=True)["config"][
+            "instance_group"
+        ]
+        return sum(instance["count"] * len(instance["gpus"]) for instance in instances)
+
+    def _load_model(self, model_name: str, timeout: int = 1000) -> None:
+        """Load a model into the server."""
+        if self.client.is_model_ready(model_name):
+            return
+
+        self.client.load_model(model_name)
+        t0 = time.perf_counter()
+        t1 = t0
+        while not self.client.is_model_ready(model_name) and t1 - t0 < timeout:
+            t1 = time.perf_counter()
+
+        if not self.client.is_model_ready(model_name):
+            raise TritonTensorRTRuntimeError(
+                f"Failed to load {model_name} on Triton in {timeout}s"
+            )
+
+    def _generate(
+        self,
+        prompts: List[str],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> LLMResult:
+        invocation_params = self._get_invocation_params(**kwargs)
+        # (WFH) TODO: It looks like we were doing this already but...?
+        self._load_model(self.model_name)
+        # stop_words = stop if stop is not None else self.stop
+        generations = []
+        # TODO: We should handle the native batching instead.
+        for prompt in prompts:
+            invoc_params = {**invocation_params, "prompt": [[prompt]]}
+            # request_id = str(random.randint(1, 9999999))  # nosec
```

Comment:
@jdye64 it doesn't look like `client.infer` requires/accepts a request_id. Could you confirm that's only used for streaming?

Reply:
Correct. The `request_id` is used to manage GRPC streams on the Triton server end so certain API calls require it. I have refactored and removed the call to `client.infer` however since we are always using the GRPC client and not HTTP client now it is safer to just always use a stream and batch locally if needed. This is required because certain models and configurations on the Triton GRCP server side might cause calls to just `infer` to fail and instead need `async_stream_infer` calls.

Reply:
helpful - thanks!